packages/apollo-gateway/src/buildQueryPlan.ts

```diff
@@ -784,17 +872,25 @@ export class QueryPlanningContext {
     return isAbstractType(type) ? this.schema.getPossibleTypes(type) : [type];
   }
 
-  getVariableUsages(selectionSet: SelectionSetNode) {
+  getVariableUsages(selectionSet: SelectionSetNode, fragments: Set<FragmentDefinitionNode>) {
```

Comment:
This is merely a matter of remembering the difference, but this is a divergence from the `getVariableUsages` [that I'm used to](https://github.com/graphql/graphql-js/blob/0eb088b3d1228ac60568912c705401341f3b769d/src/validation/ValidationContext.js#L191-L227)!

Reply:
We discussed this one via call, and I don't see another way to approach this, really. `graphql-js` has a similar function that recurses and handles full documents, however in our case we're only ever visiting a single `selectionSet` at a time. For now I'm content with the approach here, but please let me know if you have a major concern with this.